tests/unit/models/post-test.js

```diff
@@ -153,3 +153,57 @@ test('promiseTypes', function(assert) {
   assert.equal(promiseArray.constructor.superclass, ArrayProxy);
   assert.equal(promiseObject.constructor.superclass, ObjectProxy);
 });
+
+test('model action handles InvalidError', function(assert) {
+  assert.expect(6);
+
+  this.server.put('/posts/:id/publish', (request) => {
+    let data = JSON.parse(request.requestBody);
+    assert.deepEqual(data, { myParam: 'My first param' });
+    assert.equal(request.url, '/posts/1/publish');
+
+    return [422, {}, '{"errors": [{"detail": "First error", "source": {"pointer": "data"}}]}'];
+  });
+
+  let done = assert.async();
+  let payload = { myParam: 'My first param' };
+  let store = this.store();
+  let model = this.subject();
+
+  model.set('id', 1);
+  assert.equal(store.peekAll('post').get('length'), 1);
+
+  model.publish(payload).then((response) => {
```

Comment:
`model.get('errors')` is used only if there are errors during removing, updating or saving a state of the model.  So, there can be two reasons of that. First, model validation and second undefined errors.

If you are calling custom action you didn't change the state of the object. You are not saving it, removing or updating, so you are not changing his state. Of course, you can send an action which will return the model with new attributes, but it's not expected by custom action and you are not sending a whole model for that, just a few parameters.

If we will allow on assigning errors to the model we won't be able to control the lifecycle of them. When should they be cleared? After updating the record? Or after the recall? We don't even have an API for that :/

That's why I think, we should stay stateless without interacting with the model through custom actions.

So, as I just suggest higher, in my opinion, we should just add `serializedErrors` to the error object to allow you to handle that in your `catch` callback, eg:
```js
model.publish({ publisherId: 1 }).catch((error) => {
  this.set('publishErrors', error.serializedErrors);
});
```

Maybe in the future, when Ember will add public API for adding errors we will look at this topic again.